### ISSUE TITLE
The 'lint-doc' job is failing on odoc.

### ISSUE THREAD

**Issue by jonludlam** [ocurrent/ocaml-ci#991]
## Context

The 'lint-doc' job is failing on odoc, meaning we have a red cross on all ocaml-ci jobs right now.
See https://ocaml.ci.dev/github/ocaml/odoc/commit/822d266232fccdffbd4922434c81c45ab6d583f4/variant/%28lint-doc%29 as an example.

The failure seems to be happening while it's trying to build odoc-parser, and is complaining about the lack of a `dune-project` file. 

To be honest, the 'lint-doc' job is of very limited value right now, though we can do a way better job of it when we release `odoc-driver`.

**Comment by punchagan** [comment]
Thanks for the report @jonludlam! The current failure seems like a duplicate of #990 which @shonfeder is investigating. The note about switching to `odoc-driver` could be left as an issue for tackling, if you like.

**Comment by jonludlam** [comment]
I did a bit of investigation. It's failing because:

1. It pins odoc, odoc-parser, etc to /src/ without installing anything but the opam files
2. It then tries to 'opam install' odoc, which with dune 3.16 completed without error, even though it just installed a totally empty package. On dune 3.17 it complains (not unreasonably!).
3. Only then does it actually install the source.

It seems likely this is the cause of the problems in #990 too. 

Permuting the two lines (install src, then install odoc) almost works - I needed to explicitly ask to install `odoc-parser` as well as `odoc`. However, if you do this you lose an awful lot of the caching that ocaml-ci gives you. 

One possibility is to install the deps before pinning, then pin, then copy the files, then install.
